bats/tests/preferences/verify-paths.bats

```diff
@@ -0,0 +1,114 @@
+# Test case 30
+
+setup() {
+    load '../helpers/load'
+    if is_windows; then
+        skip "test not applicable on Windows"
+    fi
+    # Ensure subshells don't inherit a path that includes ~/.rd/bin
+    PATH_WITHOUT_RD=$(echo "$PATH" | tr ':' '\n' | grep -v /.rd/bin | tr '\n' ':')
+}
+
+teardown_file() {
+    load '../helpers/load'
+    run rdctl shutdown
+    assert_nothing
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start app and fix dotfiles' {
+    start_container_engine
+    wait_for_container_engine
+    ensure_dotfiles_are_completed_BUG_BUG_BUG_4519
+}
+
+# Running `bash -l -c` causes bats to hang
+@test 'bash managed' {
+    run bash -l -c "which rdctl" 3>&-
+    assert_output --partial '.rd/bin/rdctl'
+}
+
+@test 'ksh managed' {
+    if command -v ksh >/dev/null && [ -f "$HOME/.kshrc" ]; then
+        PATH="$PATH_WITHOUT_RD"
+        export PATH="$PATH_WITHOUT_RD"
+        run ksh -c "which rdctl"
+        assert_success
+        assert_output --partial '.rd/bin/rdctl'
+    else
+        skip 'ksh not found or ~/.kshrc does not exist'
+    fi
+}
+
+@test 'zsh managed' {
+    if command -v zsh >/dev/null && [ -f "$HOME/.zshrc" ]; then
+        PATH="$PATH_WITHOUT_RD"
+        export PATH="$PATH_WITHOUT_RD"
+        run zsh -i -c "which rdctl"
+        assert_success
+        assert_output --partial '.rd/bin/rdctl'
+    else
+        skip 'zsh not found or ~/.zshrc does not exist'
+    fi
+}
+
+@test 'fish managed' {
+    if command -v fish >/dev/null; then
+        PATH="$PATH_WITHOUT_RD"
+        export PATH="$PATH_WITHOUT_RD"
+        run fish -c "which rdctl"
+        assert_success
+        assert_output --partial '.rd/bin/rdctl'
+    else
+        skip 'fish not found'
+    fi
+}
+
+@test 'move to manual path-management' {
+    rdctl set --application.path-management-strategy=manual
+    sleep 5
+}
+
+@test 'bash unmanaged' {
+    PATH="$PATH_WITHOUT_RD"
+    export PATH="$PATH_WITHOUT_RD"
+    run bash -l -c "which -s rdctl" 3>&-
```

Comment:
You wrote that `which -s` is not available on Ubuntu. So this test will fail because you pass an invalid option, and not because it can't find `rdctl`. So we need to use `command -v` here as well.

Reply:
No need for `which -s`. In fact, a dev might have a symlink to `rdctl` in `/usr/local/bin`. A subshell launched by a managed-path dotrc file shouldn't find that, but when the path isn't managed, we might find that. We just want to ensure we don't pick up `~/.rd/bin/rdctl`, so that simplifies the code.